client/client_test.go

```diff
@@ -20,7 +21,8 @@ func NewClient(accountSid string, authToken string) *twilio.Client {
 	c := &twilio.Client{
 		Credentials: creds,
 		HTTPClient:  http.DefaultClient,
-		BaseURL:     "twilio.com",
+		Edge:        os.Getenv("TWILIO_EDGE"),
```

Comment:
The test shouldn't be reading these values unless it's also clearing them before each test. Otherwise tests may fail (or pass in error) when set in the testing environment.

Reply:
If I add `os.Setenv("TWILIO_EDGE", "")`, `os.Setenv("TWILIO_REGION", "")` at the end of each the region/edge tests, that should work

Reply:
[Like so](https://github.com/twilio/twilio-go/pull/62/files/6c3ee866bc0ff42a83e0675dc284e650771c69c6#diff-57cdc3e29d0ef725a50897a4fbd1c4f76353429fa66d3a785dcbc17dcd7b6f31R161)

Reply:
Sort of, but this is partially testing the logic that the test client can read from the env vars (not the actual client). I say avoid it if possible. Unless, for instance, you're testing that the actual client can read from the env vars.

Reply:
Gotcha. I was trying to test precedence, but I see that's already handled when setting `c.Region` or `c.Edge`, which is equivalent to setting those fields during instantiation of the client. I'll rip out the extraneous tests and submit a new PR

Reply:
https://github.com/twilio/twilio-go/pull/66